proof_gen/src/lib.rs

```diff
@@ -134,6 +134,18 @@
 //!     assert!(verifier_state.verify(block_proof.intern).is_ok());
 //! ```
 
+#[cfg(any(
+    all(feature = "cdk_erigon", feature = "polygon_pos"),
+    all(feature = "cdk_erigon", feature = "eth_mainnet"),
+    all(feature = "polygon_pos", feature = "eth_mainnet"),
+    not(any(
+        feature = "cdk_erigon",
+        feature = "eth_mainnet",
+        feature = "polygon_pos"
+    ))
+))]
+compile_error!("One and only one of the feature chains `cdk_erigon`, `polygon_pos` or `eth_mainnet` must be selected");
```

Comment:
Nit: you could probably put `const &str` to common crate and use the same error message everywhere?

Reply:
We can't use `common` in evm_arithmetization AFAIK because of circular dependencies though 🤔